cmd/subnetcmd/upgradecmd/generate.go

```diff
@@ -297,12 +303,15 @@ func promptContractAllowListParams(precompiles *[]params.PrecompileUpgrade, date
 	return nil
 }
 
-func promptTxAllowListParams(precompiles *[]params.PrecompileUpgrade, date time.Time) error {
+func promptTxAllowListParams(precompiles *[]params.PrecompileUpgrade, date time.Time, subnetName string) error {
 	adminAddrs, enabledAddrs, err := promptAdminAndEnabledAddresses()
 	if err != nil {
 		return err
 	}
 
+	if err = ensureAdminsHaveBalance(adminAddrs, subnetName); err != nil {
```

Comment:
This might be a big ask, but although your solution is correct, we usually try to catch things early on. 

In this scenario, if a user enters an admin address which has no balance, the user experience will be that she gets here, and hits the error, which exits the tool.

A better experience would be that on adding the new address, the balance would be immediately checked via validation.  This would immediately prevent the address from being added, give the user instant feedback, and allow her to provide a correct address right away.

The validation API is limited so not sure it will easily work, but it's worth a try!

Reply:
I think this is also a good suggestion, but probably would've been more helpful in the design/issue conception stage. Feel free to address it if you want Raymond, but you can also create a ticket for the backlog since you already have a complete solution.

Reply:
Good point - but it still could be moved earlier, e.g. when the user is selecting `Done`, _before_ the prompt UI is returning.